bin/target.ml

```diff
@@ -107,9 +107,41 @@ let resolve_path path ~(setup : Dune.Main.build_system) =
 let resolve_target common ~(setup : Dune.Main.build_system) s =
   match Alias.of_string common s ~contexts:setup.workspace.contexts with
   | Some a -> Ok [Alias a]
-  | None ->
-    let path = Path.relative Path.root (Common.prefix_target common s) in
-    resolve_path path ~setup
+  | None -> begin
+      let dep =
+        let parser =
+          Dune.Syntax.set Dune.Stanza.syntax Dune.Stanza.latest_version
+            Dune.Dune_file.Dep_conf.decode
+        in
+        Dune_lang.Decoder.parse parser Univ_map.empty
+          (Dune_lang.parse_string ~fname:"command line" ~mode:Dune_lang.Parser.Mode.Single s)
```

Comment:
Could you move this parsing into a cmdliner `Arg.conv` value? i.e. like the `path` value in `bin/arg.ml`. The idea would be to have:

```ocaml
val Arg.dep : Dune.Dune_file.Dep_conf.t Arg.conv
```

Then the error would be reported by cmdliner while parsing the command line, which seems nicer to me.

Since the parsing reports `User_error.E` exception, you will need to extract the error message from the `User_message.t` value. You can use this (from `.../sexp_tests.ml`) for that:

```ocaml
let extract_error_message (msg : User_message.t) =
  Format.asprintf "%a"
    Pp.render_ignore_tags
    (User_message.pp { msg with loc = None })
  |> String.drop_prefix ~prefix:"Error: "
  |> Option.value_exn
  |> String.trim
```

Reply:
Good idea, will do.

Reply:
This is now done.